reference/workflow_language.rst

```diff
@@ -0,0 +1,107 @@
+.. _workflow_language:
+
+Workflow Language
+=================
+
+A module consists of a module header, import statements, and definitions,
+for instance: ::
+
+  module ModuleB:0.1.0 {
+    import ModuleA:0.1.0 as A;
+    def x = A.y | A.z;
+  }
+
+An import statement includes the module to be imported (``MyModule:0.1.0``)
+and its alias (``A``).
+The alias is used to qualify functions imported from the module,
+e.g. ``A.y``.
+
+Definitions bind function names (``x``) to expressions (``A.y | A.z``).
+Any function ``x`` defined in a module ``ModuleB`` can be used in
+any other module: ::
+
+    import ModuleB:0.1.0 as B;
+    def z = filter B.x;
+
+If a name is not prefixed by a module alias, it refers to a function defined in
+the current module.
+
+Expressions combine already defined functions through the following operations:
+
+* Pipe: ``A.y | A.z``
+
+  Every value produced by ``A.y`` is passed to ``A.z``.
+
+  The output type of ``A.y`` must match the input type of ``A.z``.
+
+* Concat: ``concat A.y`` or ``A.y*``
+
+  ``A.y`` must be a function that produces lists of values,
+  in which case ``concat A.y`` is a function that "unpacks" the lists
+  and yields the same values one by one.
+
+* Filter: ``filter A.y`` or ``A.y?``
+
+  ``A.y`` must be a function that produces "optional" (potentially missing) values,
+  in which case ``filter A.y`` is a function that filters out missing values.
+
+* Type application. Some functions (notably, sources and sinks) can be specialized
```

Comment:
"notably, some sources and sinks" / "notably, the built-in sources and sinks"? I don't think sources and sinks in general have to be polymorphic (The websocket ones aren't, right?)